src/common/hooks/useInfiniteLoading.ts

```diff
@@ -1,58 +1,144 @@
+import { ActionCreatorWithoutPayload } from '@reduxjs/toolkit';
 import { PaginatedResult } from 'common/models';
-import { useEffect, useMemo, useRef, useState } from 'react';
+import { useCallback, useEffect, useMemo, useReducer } from 'react';
+import { useDispatch } from 'react-redux';
 import { UseQuery, UseQueryOptions } from 'rtk-query-config';
 
-export const useInfiniteLoading = <T, ResultType extends PaginatedResult<T>>(
-  initialUrl: string,
+export interface WithNumberIdentifier {
+  id?: number;
+}
```

Comment:
This won't be typed correctly for string identifiers (which many people like to use, such as uuid's).

Also the naming here is a little confusing, `WithIdentifier` but then saying `id` is optional?

Reply:
The reason why it is optional is that there are some models that don't have an `id` property. For example, the `HistoricalRecord`'s `id` property is `historyId`. At the moment, the changelog functionality doesn't need the remove case in the `useInfiniteLoading` hook's reducer. However, I still needed to add the `id` property somehow since that reducer case makes use of an `id` property. Should we implement something that extends these type of models so that their custom `id` property is used to fill in the value for the standard `id` property? For example, when fetching an agent's history, I could use the `useMemo` hook in order to make all fetched history items be transformed into an object that contains an `id` property whose value comes from the `historyId` property. The newly created objects wouldn't include the `historyId` property.